### PR TITLE
feat(t49): add ability to create/update player

### PR DESCRIPTION
**Pull request by AlekseySamokhin**

### CODE REVIEW

**Comment by AlekseySamokhin** [comment]
@killmenot sorry, commit -> "feat(t31): add search map component" shouldn't be here.

File: modules/core/components/common/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              <ListItemText primary={map?.name} />
+            </ListItemButton>
+          </ListItem>
+        ))}
+      </List>
+    </Box>
+  );
+};
+
+export { SearchBar };
```
Comment (killmenot) [inline modules/core/components/common/SearchBar/SearchBar.tsx:85]: a hint:
```
export const SearchBar = 
```

Comment (killmenot) [reply modules/core/components/common/SearchBar/SearchBar.tsx:85]: Let's create a task for such renaming and refactor the code in the next sprint

File: pages/players/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import { PlayerForm } from '@/modules/players/components';
+import { GET_PLAYER } from '@/modules/players/api/players';
+
+import {
+  UpdatePlayerDocument,
+  UpdatePlayerMutation,
+  UpdatePlayerMutationVariables,
+} from '@/generated/graphql';
+
+type FormValuesType = {
```
Comment (killmenot) [inline pages/players/[id].tsx:16]: this type should be part of form component

File: modules/players/components/PlayerForm/PlayerForm.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+              <Button type="submit" variant="contained" color="primary">
+                Create
+              </Button>
+            </>
+          )}
+
+          {props.onUpdatePlayer && (
+            <>
+              <Link
```
Comment (killmenot) [inline modules/players/components/PlayerForm/PlayerForm.tsx:90]: better to create a component in common for Link + Button

File: modules/core/components/common/SearchBar/SearchBar.tsx (hunk 1 above)
Comment (AlekseySamokhin) [reply modules/core/components/common/SearchBar/SearchBar.tsx:85]: Ok!

File: modules/players/components/PlayerForm/PlayerForm.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  const onSubmit = useCallback(
+    (data: FormValuesType) => {
+      if (props.onCreatePlayer) {
+        props.onCreatePlayer(data);
+
+        props.onClose();
+      }
+
+      if (props.onUpdatePlayer) {
```
Comment (killmenot) [inline modules/players/components/PlayerForm/PlayerForm.tsx:53]: isntead
```jsx
if (props.onUpdatePlayer) {
  props.onUpdatePlayer()
}

```

use this
```jsx
props.onUpdatePlayer?.()
```

Comment (killmenot) [reply modules/players/components/PlayerForm/PlayerForm.tsx:53]: onClose should not be on this level

**Review by killmenot — APPROVED** [review]

File: modules/players/components/PlayerForm/PlayerForm.tsx (hunk 4 above)
Comment (AlekseySamokhin) [reply modules/players/components/PlayerForm/PlayerForm.tsx:53]: But the buttons are in the form and when I click on add or cancel I have to hide the modal window

Comment (killmenot) [reply modules/players/components/PlayerForm/PlayerForm.tsx]: ok, lets discuss it on Wednesday also

**Pull request merged**